### ISSUE TITLE
NO DRIVER ON (toplevel/??? :  UInt[3 bits])

### ISSUE THREAD

**Issue by futurehome** [SpinalHDL/SpinalWorkshop#13]
Hi, 

I'm trying the code in PixelSolver.solutionPipelinedHyperthread, but got the errors below, it refers to id in Context defintion, anything wrong?

Thanks,
Alex

```
********************************************************************************

_NO DRIVER ON (toplevel/??? :  UInt[3 bits]), defined at
    workshop.mandelbrot.PixelSolver$Context$class.$init$(PixelSolver.scala:37)
    workshop.mandelbrot.PixelSolver$AddStageContext.<init>(PixelSolver.scala:51)

    workshop.mandelbrot.PixelSolver$$anon$4$$anonfun$8.apply(PixelSolver.scala:93)
    spinal.lib.Flow.<init>(Flow.scala:22)
    spinal.lib.FlowFactory.apply(Flow.scala:10)
    spinal.lib.Flow.clone(Flow.scala:24)
    spinal.lib.Flow.clone(Flow.scala:20)__

```

**Comment by Dolu1990** [comment]
Hi,

Thanks :D
Fixed now with https://github.com/SpinalHDL/SpinalWorkshop/commit/4bf48200c3f84dd27a6a6a6ee04aebe36c39be45

Charles

**Comment by futurehome** [comment]
Thanks for the fixing.
Why doesn't it work with trait?

Alex

> Hi,
> 
> Thanks :D
> Fixed now with [4bf4820](https://github.com/SpinalHDL/SpinalWorkshop/commit/4bf48200c3f84dd27a6a6a6ee04aebe36c39be45)
> 
> Charles

**Comment by Dolu1990** [comment]
I did some changes in the way SpinalHDL identify elements of Bundles (was done via scala reflection, now done via a scala compiler plugin), apparently, trait were not part of it anymore (didn't knew about it XD)

So basicaly, the Bundle where missing elements (from the SpinalHDL vew)
